apps/studio/data/projects/project-detail-query.ts

```diff
@@ -1,18 +1,34 @@
 import { QueryClient, useQuery, UseQueryOptions } from '@tanstack/react-query'
 
-import { get, isResponseOk } from 'lib/common/fetch'
-import { API_URL } from 'lib/constants'
-import { Project, ResponseError } from 'types'
+import { get } from 'data/fetchers'
+import { ResponseError } from 'types'
 import { projectKeys } from './keys'
+import { components } from 'data/api'
 
 export type ProjectDetailVariables = { ref?: string }
 
+export type ProjectMinimal = components['schemas']['ProjectInfo']
+export type ProjectDetail = components['schemas']['ProjectDetailResponse']
+
+export interface Project extends ProjectDetail {
+  /**
+   * postgrestStatus is available on client side only.
+   * We use this status to check if a project instance is HEALTHY or not
+   * If not we will show ConnectingState and run a polling until it's back online
+   */
+  postgrestStatus?: 'ONLINE' | 'OFFLINE'
```

Comment:
> available on client side only

Where is it being added in?

Reply:
when we call `setProjectPostgrestStatus` from projects-query